src/ClimaAtmos/Dycore/src/VanillaAtmosDiscretizations.jl

```diff
@@ -191,41 +195,59 @@ step
 
     This estimate is currently very conservative, needs to be revisited
 """
-function estimatedt(disc::VanillaAtmosDiscretization{T, dim, N, Np, DA},
-                    Q::AtmosStateArray) where {T, dim, N, Np, DA}
+function estimatedt(disc::VanillaAtmosDiscretization{T, dim, N, Np, DA, nmoist},
+                    Q::AtmosStateArray) where {T, dim, N, Np, DA, nmoist}
   @assert T == eltype(Q)
   G = disc.grid
   vgeo = G.vgeo
   # FIXME: GPUify me
   host_array = Array ∈ typeof(Q).parameters
   (h_vgeo, h_Q) = host_array ? (vgeo, Q) : (Array(vgeo), Array(Q))
-  estimatedt(Val(dim), Val(N), G, disc.gravity, h_Q, h_vgeo, G.topology.mpicomm)
+  estimatedt(Val(dim), Val(N), Val(nmoist), G, disc.gravity, h_Q, h_vgeo, G.topology.mpicomm)
 end
 
 # FIXME: This needs cleaning up
-function estimatedt(::Val{dim}, ::Val{N}, G, gravity, Q, vgeo,
-                    mpicomm) where {dim, N}
+function estimatedt(::Val{dim}, ::Val{N}, ::Val{nmoist}, G, gravity, Q, vgeo,
+                    mpicomm) where {dim, N, nmoist}
 
   DFloat = eltype(Q)
 
   Np = (N+1)^dim
   (_, _, nelem) = size(Q)
 
   dt = [floatmax(DFloat)]
+  
+  # Allocate 3 spaces for moist tracers qm, with a zero default value
+  q_m = zeros(DFloat, max(3, nmoist))
 
   if dim == 2
     @inbounds for e = 1:nelem, n = 1:Np
       ρ, U, V = Q[n, _ρ, e], Q[n, _U, e], Q[n, _V, e]
       E = Q[n, _E, e]
       y = vgeo[n, G.yid, e]
-      P = gdm1*(E - (U^2 + V^2)/(2*ρ) - ρ*gravity*y)
-
+      
+      #Compute Temperature and Internal Energy per unit mass
+      #Obtain moist variables from state vector
+      for m = 1:nmoist
+          s = _nstate + m
+          q_m[m] = Q[n, s, e]/ρ
+      end
+      
+        e_int = E / ρ - ((U^2 + V^2)/(2*ρ) + ρ * gravity * y) / ρ
+        T = MoistThermodynamics.saturation_adjustment(e_int, ρ, q_m[1])
```

Comment:
Do we need the `MoistThermodynamics.` prefix? What’s our policy here? @simonbyrne?

Reply:
No - since we export functions from MoistThermodynamics; but I've left it in for clarity for now

Reply:
My preference would be to remove it to reduce redundancy, unless others find it helpful to have. It'd be nice to be consistent about this across the code.